chore: fix biome and lint warnings

Rodriguespn opened the pull request:
## Changes Made

This PR addresses multiple linting and code quality improvements:

### 1. Non-null Assertion Fixes
- **Files**: Multiple hook files in `src/web/hooks/`
- Replaced all non-null assertions (`!`) with explicit undefined checks and descriptive error messages

### 2. TypeScript Type Safety
- **Files**: `src/test/widget.test.ts`, `src/web/data-llm.test.tsx`
- Replaced `any` types with proper TypeScript types

### 3. CI/CD Improvements
- **Files**: `.github/workflows/ci.yml`, `package.json`, `.nvmrc`
- Created `.nvmrc` file using `lts/krypton` to avoid locking to a specific minor/patch version
- Separated code quality checks from test job for better CI feedback:
  - **`code-quality` job**: Runs `biome ci .` (formatting and linting) and TypeScript type checking independently
  - **`test` job**: Builds and runs unit tests independently
- Updated CI to use `biome ci` instead of `biome check` following [Biome's CI documentation](https://biomejs.dev/recipes/continuous-integration/) for better GitHub Actions integration (annotations)
- Simplified `test` script to only run unit tests; code quality checks are handled by the separate CI job

### 4. Biome Extension Configuration
- **File**: `.vscode/settings.json` (if applicable)
- Configured Biome extension to work properly in Cursor IDE

## Rationale

The separation of code quality and test jobs allows:
- **Non-blocking tests**: Tests can run even if TypeScript has errors, providing faster feedback during development when you might have incomplete type definitions but still want to verify runtime behavior
- **Better CI integration**: Using `biome ci` provides GitHub Actions annotations for better visibility of code quality issues
- **Clear separation of concerns**: Code quality checks (formatting, linting, type checking) are TypeScript-specific and run independently from functional tests

Closes: https://github.com/alpic-ai/skybridge/issues/83
Closes: https://github.com/alpic-ai/skybridge/issues/84

Rodriguespn commented on .github/workflows/ci.yml:
```diff
@@ -6,6 +6,31 @@ on:
     branches: [main]
 
 jobs:
+  code-quality:
```
Let's continue this discussion [here](https://github.com/alpic-ai/skybridge/pull/90#issuecomment-3650739010)

fredericbarthelet commented on package.json:
```diff
@@ -20,6 +20,7 @@
     "build": "tsc && pnpm run build:templates",
     "build:templates": "cp -r src/server/templates dist/src/server/",
     "format": "biome check --write",
+    "lint": "biome lint --error-on-warnings",
```
It was removed, this is now resolved :)

fredericbarthelet commented on package.json:
```diff
@@ -20,10 +20,9 @@
     "build": "tsc && pnpm run build:templates",
     "build:templates": "cp -r src/server/templates dist/src/server/",
     "format": "biome check --write",
-    "test": "pnpm run test:unit && pnpm run test:type && pnpm run test:format",
```
please keep this in order to be able to run locally and iterate on a specific dev

Rodriguespn commented:
> > Regarding biome lint script: Removed it frompackage.json since biome check already includes linting.
> 
> Perfect 👌
> 
> > Regarding separate code-quality job: I separated it for two reasons: non-blocking test and biome ci
> 
> My believes on that matter is that CI is an automated shared quality check to ensure no merging occurs on a broken PR. Feedback loop on types correctness and unit tests pass should be done locally on your IDE while developing the feature, not once you push to Github :) However I'm open to give better visibility with separated jobs that will all always run in parallel. But I'd like to keep it DRY (so no double installs) and as simple as possible. Hereafter is a suggested implementation where install is still made once, and 3 jobs are separated using a matrix. WDYT ?
> 
> switching from `biome check` to `biome ci` is a good improvement, but unrelated to jobs running in series or in parallel. You can however update the command in the package.json
> 
> ```yaml
> name: CI
> 
> on:
>   pull_request:
>   push:
>     branches: [main]
> 
> jobs:
>   install:
>     name: Install
>     runs-on: ubuntu-latest
>     steps:
>       - uses: actions/checkout@v4
> 
>       - uses: pnpm/action-setup@v4
> 
>       - uses: actions/setup-node@v4
>         with:
>           node-version: .nvmrc
>           cache: 'pnpm'
> 
>       - run: pnpm install --frozen-lockfile
> 
>       - uses: actions/cache/save@v4
>         with:
>           path: |
>             node_modules
>             dist
>           key: build-${{ github.sha }}
> 
>   test:
>     name: ${{ matrix.name }}
>     needs: install
>     runs-on: ubuntu-latest
>     strategy:
>       fail-fast: false
>       matrix:
>         include:
>           - name: Unit Tests
>             command: test:unit
>           - name: Type Check
>             command: test:type
>           - name: Lint & Format
>             command: test:format
>     steps:
>       - uses: actions/checkout@v4
> 
>       - uses: pnpm/action-setup@v4
> 
>       - uses: actions/setup-node@v4
>         with:
>           node-version: .nvmrc
>           cache: 'pnpm'
> 
>       - uses: actions/cache/restore@v4
>         with:
>           path: |
>             node_modules
>             dist
>           key: build-${{ github.sha }}
> 
>       - run: pnpm run ${{ matrix.command }}
>       ```
> ```

Agreed on CI being for shared quality checks, not developer feedback loops. The separate jobs are mainly for better visibility during PR reviews - when a check fails, it's immediately clear whether it's tests, types, or formatting without having depencies between them. I understand that the test suit is not big at the moment to run locally but as it grows, it will come handy to run it in the CI without being blocked by lint and formatting. Happy to go with my suggestion or your proposal nevertheless.

I've implemented your matrix approach with one change: the build runs in the install job so the `dist` folder is cached and available to all test jobs. The sequential `test` script is kept in package.json for local development.

Here's the updated `.github/workflows/ci.yml`:

```yaml
name: CI

on:
  pull_request:
  push:
    branches: [main]

jobs:
  install:
    name: Install
    runs-on: ubuntu-latest
    permissions:
      contents: read
    steps:
      - uses: actions/checkout@v4

      - uses: pnpm/action-setup@v4

      - uses: actions/setup-node@v4
        with:
          node-version-file: ".nvmrc"
          cache: "pnpm"

      - name: Install dependencies
        run: pnpm install --frozen-lockfile

      - name: Build project
        run: pnpm run build

      - uses: actions/cache/save@v4
        with:
          path: |
            node_modules
            dist
          key: build-${{ github.sha }}

  test:
    name: ${{ matrix.name }}
    needs: install
    runs-on: ubuntu-latest
    permissions:
      contents: read
    strategy:
      fail-fast: false
      matrix:
        include:
          - name: Unit Tests
            command: test:unit
          - name: Type Check
            command: test:type
          - name: Lint & Format
            command: test:format # Runs "biome ci ."
    steps:
      - uses: actions/checkout@v4

      - uses: pnpm/action-setup@v4

      - uses: actions/setup-node@v4
        with:
          node-version-file: ".nvmrc"
          cache: "pnpm"

      - uses: actions/cache/restore@v4
        with:
          path: |
            node_modules
            dist
          key: build-${{ github.sha }}

      - name: Build project
        if: matrix.name == 'Unit Tests'
        run: pnpm run build

      - name: Run ${{ matrix.name }}
        run: pnpm run ${{ matrix.command }}
```

WDYT about this implementation?

fredericbarthelet reviewed (changes requested):
Thanks for udpating your PR @Rodriguespn  :)

A few final comments and we should be good to go.

Rodriguespn commented on src/web/hooks/use-display-mode.ts:
```diff
@@ -2,7 +2,12 @@ import type { DisplayMode } from "../types.js";
 
 export function useDisplayMode() {
-  const displayMode = useOpenAiGlobal("displayMode")!;
+  const displayMode = useOpenAiGlobal("displayMode");
```
Apologies for the confusion. I misunderstood your suggestion at first.
You're right, the current check only logs a warning, and the optional chaining in the getSnapshot callback allows silent failures. I'll apply your recommendation and move the error to the external store getter.

We can continue discussing the unified useOpenAiGlobal hook for properties in #94.

fredericbarthelet replied:
Thanks for updating with this strategy @Rodriguespn :)
I don't quite get the necessity for this new required options and overload.
`toolOuput` and `toolResponseMetadata` already have a typing representing a loading state and a loaded state.
I took the liberty to push a new commit to show what I had in mind. Let me know what you think and feel free to merge if you like my proposal, or push another commit if you have a better idea :)

fredericbarthelet commented on .github/workflows/ci.yml:
```diff
@@ -6,25 +6,65 @@ on:
+
       - name: Build project
+        if: matrix.name == 'Unit Tests'
         run: pnpm run build
```
Since tsc is the builder and test:type tests `tsc`, we should be good to go for the time being
